src/tools/extract-conditionals.cpp

```diff
@@ -0,0 +1,991 @@
+#include "util/conditional_restrictions.hpp"
+#include "util/exception.hpp"
+#include "util/for_each_pair.hpp"
+#include "util/log.hpp"
+#include "util/opening_hours.hpp"
+#include "util/version.hpp"
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/fusion/include/adapt_adt.hpp>
+#include <boost/geometry.hpp>
+#include <boost/geometry/index/rtree.hpp>
+#include <boost/program_options.hpp>
+#include <boost/spirit/include/qi.hpp>
+#include <boost/spirit/include/support_line_pos_iterator.hpp>
+
+#include <osmium/handler.hpp>
+#include <osmium/index/map/sparse_mem_array.hpp>
+#include <osmium/io/any_input.hpp>
+#include <osmium/tags/regex_filter.hpp>
+#include <osmium/visitor.hpp>
+
+#include <shapefil.h>
+
+#include <fstream>
+#include <iostream>
+#include <unordered_map>
+#include <unordered_set>
+
+#include <chrono>
+#include <cstdlib>
+//#include <ctime>
+
+// better place in third_party/libosmium/include/osmium/tags/regex_filter.hpp
+namespace osmium
+{
+
+namespace tags
+{
+
+template <> struct match_key<std::regex>
+{
+    bool operator()(const std::regex &rule_key, const char *tag_key)
+    {
+        return std::regex_match(tag_key, rule_key);
+    }
+}; // struct match_key<std::regex>
+}
+}
```

Comment:
I think we can update libosmium in a independent pr, merge into master, rebase and then remove this.

https://github.com/osmcode/libosmium/releases

Check scripts/update_dependencies.